### PR TITLE
ConfigurationGathering: Add strict mode

### LINKED ISSUE #3987: Add strict .coafile mode
Some of coala warnings indicate .coafile has problems.
Warnings are also useful in TTY mode, but have lesser value in CI mode.
Some CI users might like to have only warnings for config related problems, such as missing bears, so that their CI doesnt turn red regularly, and the rest of their CI works.
Other CI users would see a config error as a regression that must be fixed. so a strict mode cant equal CI mode.

Opened via [gitter](https://gitter.im/coala/coala/?at=58d58df15231ebb865b24ae8) by @jayvdb

### PR DESCRIPTION
**Pull request by RJ722**
If coala runs with --strict mode, it should throw up an error if
it does not finds a .coafile.

Closes #3987

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [X] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [X] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by RJ722** [comment]
@sils @Makman2 @jayvdb Is this approach feasible?

**Comment by RJ722** [comment]
>In commit if it does not find a .coafile.

@arjunsinghy96 sorry, I don't understand what are you saying?

**Comment by arjunsinghy96** [comment]
I meant to say in commit message `finds` -> `find`

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -53,25 +53,27 @@ def load_config_file(filename, log_printer, silent=False):
...
     """
     filename = os.path.abspath(filename)
-
+    strict = True  # TODO: Remove it (Just for testing purposes)
     try:
         return ConfParser().parse(filename)
     except FileNotFoundError:
         if not silent:
-            if os.path.basename(filename) == Constants.default_coafile:
+            if (not strict) and \
```
Comment (Siddhartha1234) [inline coalib/settings/ConfigurationGathering.py]: This wasn't the point of adding a "strict mode". It was to enable some warnings like glob pattern not matching any file as errors. I don't see any warning -> error changes. Please discuss in gitter if you need further insight before submitting a PR.

**Comment by jayvdb** [comment]
Not much of a WIP here.
Closing and unassigning.

**Pull request closed without merging**